### ISSUE TITLE
Hidden fields and conditional fields

### ISSUE THREAD

**Issue by Betree** [docusealco/docuseal#168]
Hi,

We are looking into DocuSeal to see if we could replace our current implementation of Dropbox Forms (ex-HelloSign/HelloWorks) for https://opencollective.com. 

So far, the form builder works pretty well, but it seems to be missing an important feature for our use case: in our current implementation, we have defined a special question that is not displayed on the PDF document: `Are you US Person or Entity?` (Yes/No). Based on the answer provided, we'll either show the W9 fields (if it's an organization) or the W-8 BEN fields (if it's an individual):

![image](https://github.com/docusealco/docuseal/assets/1556356/aedf24ea-3484-4b36-b0a5-bd820e6d9e03)

In summary, what we need is:
1. The ability to add Hidden Fields to collect information that is not displayed on the document
2. The ability to define Conditional Fields that will only be displayed if a previous field is set to a certain value

Is that something you have on your radar? 

Thanks in advance!

**Comment by omohokcoj** [comment]
@Betree 

>The ability to add Hidden Fields to collect information that is not displayed on the document

It's already available - you can just click on the field type to add it to the document form but without placing the field area on the page (aka hidden field)
![image](https://github.com/docusealco/docuseal/assets/5418788/6fbd3769-d931-4419-b46f-d9f2fcae5d57)

>The ability to define Conditional Fields that will only be displayed if a previous field is set to a certain value

Currently we don't have conditional fields support - but it seems like in your case you just need a single condition for US vs not US to switch between W9 and W-8 BEN forms - looks like this might be just two different PDF form templates?
In this case you can just develop this condition in your SaaS and send W9 or W-8 BEN template forms based on the user answer.

Also, could you please elaborate more regarding your use case for conditional fields (in case you also need them per other template form fields) - this might help us to figure out the best way to implement this in the tool.

**Comment by Betree** [comment]
>It's already available - you can just click on the field type to add it to the document form but without placing the field area on the page (aka hidden field)

Perfect, thanks for the pointer!

> Currently we don't have conditional fields support - but it seems like in your case you just need a single condition for US vs not US to switch between W9 and W-8 BEN forms - looks like this might be just two different PDF form templates?

We could indeed do it that way, that's a feasible workaround for us.

> Also, could you please elaborate more regarding your use case for conditional fields (in case you also need them per other template form fields) - this might help us to figure out the best way to implement this in the tool.

That's it really, I've simplified a bit the example above but our full use case is to switch between W9, W-8 BEN, and W-8 BEN-E forms based on whether you whether you're a US person/entity, and whether you're an individual or a business/entity.

**Comment by omohokcoj** [comment]
@Betree thank you for those details - looks like just two separate template forms might work best for your use case.
(anyway conditional fields are in the roadmap, we just need to collect a bit more real-life use cases and feedback from interested users to find the best way to implement it).
Let me know if you have any other questions or if we can close the issue.

**Comment by Betree** [comment]
I'm good with closing the issue, thanks for the feedback!
